dotCMS/src/main/java/com/dotmarketing/portlets/folders/business/FolderFactoryImpl.java

```diff
@@ -107,88 +100,81 @@ protected Folder find(String folderInode) throws DotDataException {
+			throws DotStateException, DotDataException {
 		Identifier id = APILocator.getIdentifierAPI().find(folder);
-		HibernateUtil dh = new HibernateUtil(Folder.class);
-		String condition = "";
-		if(UtilMethods.isSet(showOnMenu)){
-			condition = "show_on_menu = " + com.dotmarketing.db.DbConnectionFactory.getDBTrue();
-		}
-		dh.setSQLQuery("SELECT {folder.*} from folder folder, inode folder_1_, identifier identifier where folder.identifier = identifier.id and "
-				+ "folder_1_.type = 'folder' and folder_1_.inode = folder.inode and identifier.parent_path = ? and identifier.host_inode = ? and "
-				+ condition + " order by sort_order, name");
-		dh.setParam(id.getPath());
-		dh.setParam(id.getHostId());
-		return (java.util.List<Folder>)dh.list();
-
+		return getSubFolders(showOnMenu, id.getPath(), id.getHostId(), null);
 	}
+
 	@SuppressWarnings("unchecked")
 	@Override
-	protected List<Folder> findSubFolders(Host host, boolean showOnMenu) throws DotHibernateException   {
+	protected List<Folder> findSubFolders(final Host host, Boolean showOnMenu)
+			throws DotHibernateException {
+
+		return getSubFolders(showOnMenu, "/", host.getIdentifier(), null);
+	}
 
-		HibernateUtil dh = new HibernateUtil(Folder.class);
+	private List<Folder> getSubFolders(Boolean showOnMenu, String path, final String hostId, String order) {
+		DotConnect dc    = new DotConnect();
 		String condition = "";
-		if(UtilMethods.isSet(showOnMenu)){
-			condition = "show_on_menu = " + com.dotmarketing.db.DbConnectionFactory.getDBTrue();
+		StringBuilder orderBy   = new StringBuilder(" order by ");
+		StringBuilder query = new StringBuilder();
+
+		if (UtilMethods.isSet(showOnMenu)) {
+			condition = "show_on_menu = " + (showOnMenu ? DbConnectionFactory
+					.getDBTrue() : DbConnectionFactory.getDBFalse());
 		}
-		String query = "select {folder.*} from folder, inode folder_1_, identifier identifier where parent_path = '/' and "+
-		               "folder_1_.type = 'folder' and folder.inode = folder_1_.inode and folder.identifier = identifier.id and host_inode = ? and "
-					   + condition + " order by sort_order, name";
 
-		dh.setSQLQuery(query);
-		dh.setParam(host.getIdentifier());
+		if (!UtilMethods.isSet(order)){
+			orderBy.append("sort_order, name");
+		}else{
+			orderBy.append(order);
+		}
+
+		query.append("SELECT folder.* from folder folder, inode folder_1_, identifier identifier ").
+				append("where folder.identifier = identifier.id and ").
+				append("folder_1_.type = 'folder' and folder_1_.inode = folder.inode and ").
+				append("identifier.parent_path = ? and identifier.host_inode = ? ");
+
+		query.append((!condition.isEmpty()?" and " + condition:condition) + orderBy);
+
+		dc.setSQL(query.toString());
+		dc.addParam(path);
+		dc.addParam(hostId);
 
-		return (java.util.List<Folder>)dh.list();
+		try{
+			return ConvertToPOJOUtil.convertDotConnectMapToFolder(dc.loadResults());
+		}catch(Exception e){

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)